### PR TITLE
Improve the issue template.

### PR DESCRIPTION
**Pull request by jamestalmage**
The current issue template uses HTML comment tags to remove template text from the rendered output. I have gone through the last dozen issues filed, and it appears everyone deletes that anyways. By removing the comment tags, people can click "Preview" and get a pretty print of the template markdown.

Reformat lists and links. Tries to get links as close to the left edge as possible, so they are easy to see, and copy-paste from the raw markdown editor. Puts multiple links on their own line for the same reason.

Adds a request that they explicitly tell us what their AVA and `npm` versions are.

Adds a section for their AVA config and CLI args.

Adds a section for linking to thier project if public.

Requests they create a minimal reproduction and post it to GitHub.

### CODE REVIEW

File: .github/issue_template.md [hunk 1]
```diff
@@ -1,33 +1,70 @@
...
 
 ```
 node -e "var os=require('os');console.log('Node.js ' + process.version + '\n' + os.platform() + ' ' + os.release())"
 ```
 
--->
+The versions of AVA and `npm` in use:
+
+```
+ava --version
```
Comment (sindresorhus) [inline .github/issue_template.md]: This is not really necessary as we already tell them to reproduce it with the latest version.

Comment (jamestalmage) [inline .github/issue_template.md]: Yes. But getting them to actually state "I'm running version XXX" eliminates one question-answer round. When trying to figure out the problem

Comment (sindresorhus) [inline .github/issue_template.md]: Good point.

File: .github/issue_template.md [hunk 2]
```diff
@@ -1,33 +1,70 @@
-<!--
+<!-- CLICK "Preview" FOR MORE READABLE FORMAT -->
 
 ## Prerequisites
 
 - Read the [contributing guidelines](https://github.com/sindresorhus/ava/blob/master/contributing.md).
-- Support questions are better asked in our [chat](https://gitter.im/sindresorhus/ava) or on [Stack Overflow](https://stackoverflow.com/questions/tagged/ava).
+- Support questions are better asked in one of the following locations
```
Comment (sindresorhus) [inline .github/issue_template.md]: Since you're not hiding these with HTML comments anymore we can change them back to checkboxes;

`- ` => `- [ ]`

Comment (sindresorhus) [inline .github/issue_template.md]: `locations` => `locations:`

File: .github/issue_template.md [hunk 3]
```diff
@@ -1,33 +1,70 @@
...
+  - [our chat](https://gitter.im/sindresorhus/ava)
```
Comment (sindresorhus) [inline .github/issue_template.md]: `our` => `Our`

File: .github/issue_template.md [hunk 4]
```diff
@@ -1,33 +1,70 @@
...
+  - [Stack Overflow](https://stackoverflow.com/questions/tagged/ava)
 - Ensure the issue isn't already reported.
-- Should be reproducible with the latest AVA version. (Ensure `ava --version` matches ![](https://img.shields.io/npm/v/ava.svg))
-
--->
+- Should be reproducible with the latest AVA version.
+  - (Ensure `ava --version` matches ![](https://img.shields.io/npm/v/ava.svg))
```
Comment (sindresorhus) [inline .github/issue_template.md]: Maybe add a HTML comment to tell them to remove this part when done.

File: .github/issue_template.md [hunk 5]
```diff
@@ -1,33 +1,70 @@
...
+```js
+// Avoid posting hundreds of lines of source code.
+// Edit to just the relevant portions.
+```
 
+### Stack Trace
 
-## Environment
+```
+COPY THE FAILURE STACK TRACE HERE
```
Comment (sindresorhus) [inline .github/issue_template.md]: ERROR AND STACK TRACE

We just got a bug report without the actual error: 

```
ava/lib/serialize-error.js:11
        target.stack = beautifyStack(source.stack);
```

File: .github/issue_template.md [hunk 6]
```diff
@@ -1,33 +1,70 @@
...
+```
+
+### Configuration
```
Comment (sindresorhus) [inline .github/issue_template.md]: Configuration => Config

File: .github/issue_template.md [hunk 7]
```diff
@@ -1,33 +1,70 @@
...
+
+Copy the relevant section from `package.json`:
+
+```json
+{
+  "ava": {
+  ...
+  }
+}
```
Comment (sindresorhus) [inline .github/issue_template.md]: We should automatically include this in the `--report` command

File: .github/issue_template.md [hunk 8]
```diff
@@ -1,33 +1,70 @@
...
+```
+
+### Command Line Arguments
 
-<!--
+Copy your npm build scripts or the `ava` command used:
+
+```
+ava [OPTIONS HERE]
+```
```
Comment (sindresorhus) [inline .github/issue_template.md]: This too, re `--report`

File: .github/issue_template.md (hunk 1 above)
Comment (sindresorhus) [inline .github/issue_template.md]: I would include these in the above `Environment` snippet though.

File: .github/issue_template.md (hunk 2 above)
Comment (jamestalmage) [inline .github/issue_template.md]: I considered that, but you can't check them off in Preview mode, and we really do want them to delete the instructions stuff as they complete them, right? I thought people might be tempted to leave the check boxes there.

Comment (sindresorhus) [inline .github/issue_template.md]: 👍

**Comment by jfmengels** [comment]
I think the tone could be warmer, so that it feel less like you're going to get yelled at if you missed a step. You guys are all really nice and helpful, but that doesn't convey here :)

**Comment by jamestalmage** [comment]
> I think the tone could be warmer,

Yeah, I struggled with that. I don't want to add a lot of verbosity getting there though. It's already pretty long, and I don't want to add flowery language that doesn't help them just get on with it.

I'm sure there are ways to improve it that would keep it short and sweet though. I'm open to suggestions.

**Comment by sindresorhus** [comment]
@jfmengels Any specific suggestions would help ;)

**Comment by jfmengels** [comment]
> Any specific suggestions would help ;)

Yeah I know, sorry about the lack of specificity. Will try to come up with stuff later tonight, but it's not my forte either.

In the meantime and for reading purposes, there is @benmosher's very nice contributing file https://github.com/benmosher/eslint-plugin-import/pull/258. It's not the same topic, but it can be a source of inspiration (which I'll probably re-read before thinking of improving this).

**Comment by jamestalmage** [comment]
OK, I incorporated the feedback, and tried to make it a little less intimidating with https://github.com/avajs/ava/pull/787/commits/4edabb9895425cbe3ec0de5b11c964748d6e8259

File: .github/issue_template.md [hunk 9]
```diff
@@ -1,33 +1,68 @@
-<!--
+<!-- CLICK "Preview" FOR INSTRUCTIONS IN A MORE READABLE FORMAT -->
 
 ## Prerequisites
 
+- We realize there is a lot of data requested here. We ask only that you do your best to provide as much information as possible so we can better help you.
 - Read the [contributing guidelines](https://github.com/sindresorhus/ava/blob/master/contributing.md).
-- Support questions are better asked in our [chat](https://gitter.im/sindresorhus/ava) or on [Stack Overflow](https://stackoverflow.com/questions/tagged/ava).
+- Support questions are better asked in one of the following locations:
+  - [Our chat](https://gitter.im/sindresorhus/ava)
```
Comment (jfmengels) [inline .github/issue_template.md]: Link has changed: https://gitter.im/avajs/ava

**Comment by jfmengels** [comment]
LGTM, except for the Gitter link. Sorry for blocking this :s

File: .github/issue_template.md [hunk 10]
```diff
@@ -1,33 +1,68 @@
-<!--
+<!-- CLICK "Preview" FOR INSTRUCTIONS IN A MORE READABLE FORMAT -->
 
 ## Prerequisites
 
+- We realize there is a lot of data requested here. We ask only that you do your best to provide as much information as possible so we can better help you.
```
Comment (jfmengels) [inline .github/issue_template.md]: It's nice that this is said explicitly :+1:

**Comment by sindresorhus** [comment]
Very good improvement. Thanks James.

I manually merged and fixed @jfmengels comment.

**Pull request closed without merging**